Source/Core/DolphinQt/Achievements/AchievementProgressWidget.cpp

```diff
@@ -51,6 +51,37 @@ AchievementProgressWidget::CreateAchievementBox(const rc_api_achievement_definit
 {
   if (!AchievementManager::GetInstance()->IsGameLoaded())
     return new QGroupBox();
+  QLabel* a_badge = new QLabel();
+  const auto unlock_status = AchievementManager::GetInstance()->GetUnlockStatus(achievement->id);
+  auto badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.locked_badge);
```

Comment:
`const_cast` is nonsense here, `const AchievementManager::BadgeStatus* badge = &unlock_status.locked_badge` works just fine.

In general, `const_cast` has pretty much exactly one valid use case: You're passing an object to an API that wants it as non-const but you know that the object is not going to be modified by the thing you're passing it to, the API is just wrong/sloppy. That's it. It's practically never useful.